js/tab.js

```diff
@@ -134,15 +134,24 @@ define([
   
   Tab.prototype.detectSyntax = function(userConfig) {
     //settings are async
-    var self = this;
-    Settings.pull("user").then(function(data) {
+    Settings.pull("user").then((data) => {
       var userConfig = data.user;
-      self.setUseSoftTabs(!userConfig.useTabs);
-      self.setTabSize(userConfig.indentation || 2);
-      self.setUseWrapMode(userConfig.wordWrap);
-      self.setWrapLimit(userConfig.wrapLimit || null);
-      self.setNewLineMode(userConfig.lineEnding || "auto");
-      self.setUseWorker(userConfig.useWorker);
+      this.setUseSoftTabs(!userConfig.useTabs);
+      this.setTabSize(userConfig.indentation || 2);
+      this.setUseWrapMode(userConfig.wordWrap);
+      this.setWrapLimit(userConfig.wrapLimit || null);
+      this.setNewLineMode(userConfig.lineEnding || "auto");
+      this.setUseWorker(userConfig.useWorker);
+      
+      if (userConfig.useWorker && userConfig.aceWorkerOptions) {
```

Comment:
That makes sense. What would you recommend the options key be?
Some thoughts:
`"javascriptWorker":{ "asi":true }`
`"worker":{ "javascript":{ "asi":true }}`
`"jsHint":{ "asi":true }`